### PR TITLE
fixes #1338: Always check skipped terms after attempting to traverse …

### LINKED ISSUE #1338: ContentOrderedEvaluator not processing skips

### PR DESCRIPTION
**Pull request by FineAndDandy**
…if not done

### CODE REVIEW

File: warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentFunctionsTest.java [hunk 1]
```diff
@@ -180,6 +180,86 @@ public void testEvaluation1() {
...
+        t3 = asList(Arrays.asList(1, 101, 202, 213, 253, 312, 336), Arrays.asList(0, 0, 0, 0, 0, 0, 0));
+        
+        termOffSetMap.put("a", new TermFrequencyList(Maps.immutableEntry(new Zone("CONTENT", true, eventId), t1)));
+        termOffSetMap.put("b", new TermFrequencyList(Maps.immutableEntry(new Zone("CONTENT", true, eventId), t2)));
+        termOffSetMap.put("c", new TermFrequencyList(Maps.immutableEntry(new Zone("CONTENT", true, eventId), t3)));
+        
+        context.set(Constants.TERM_OFFSET_MAP_JEXL_VARIABLE_NAME, termOffSetMap);
+        Object o = expr.evaluate(context);
+        
+        Assert.assertTrue(expect(o, true));
```
Comment (apmoriarty) [inline warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentFunctionsTest.java:200]: Let's add a comment saying that "offset 252, 252, 253 evaluates to true"

File: warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentFunctionsTest.java [hunk 2]
```diff
@@ -180,6 +180,86 @@ public void testEvaluation1() {
...
+        t3 = asList(Arrays.asList(1, 101, 202, 213, 251, 312, 336), Arrays.asList(0, 0, 0, 0, 0, 0, 0));
+        
+        termOffSetMap.put("a", new TermFrequencyList(Maps.immutableEntry(new Zone("CONTENT", true, eventId), t1)));
+        termOffSetMap.put("b", new TermFrequencyList(Maps.immutableEntry(new Zone("CONTENT", true, eventId), t2)));
+        termOffSetMap.put("c", new TermFrequencyList(Maps.immutableEntry(new Zone("CONTENT", true, eventId), t3)));
+        
+        context.set(Constants.TERM_OFFSET_MAP_JEXL_VARIABLE_NAME, termOffSetMap);
+        Object o = expr.evaluate(context);
+        
+        Assert.assertTrue(expect(o, true));
```
Comment (apmoriarty) [inline warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentFunctionsTest.java:220]: Comment indicating why this is a reverse and not a forward match?

File: warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentFunctionsTest.java [hunk 3]
```diff
@@ -180,6 +180,86 @@ public void testEvaluation1() {
...
+        t3 = asList(Arrays.asList(1, 101, 202, 213, 252, 312, 336), Arrays.asList(0, 0, 0, 0, 0, 0, 0));
+        
+        termOffSetMap.put("a", new TermFrequencyList(Maps.immutableEntry(new Zone("CONTENT", true, eventId), t1)));
+        termOffSetMap.put("b", new TermFrequencyList(Maps.immutableEntry(new Zone("CONTENT", true, eventId), t2)));
+        termOffSetMap.put("c", new TermFrequencyList(Maps.immutableEntry(new Zone("CONTENT", true, eventId), t3)));
+        
+        context.set(Constants.TERM_OFFSET_MAP_JEXL_VARIABLE_NAME, termOffSetMap);
+        Object o = expr.evaluate(context);
+        
+        Assert.assertTrue(expect(o, true));
```
Comment (apmoriarty) [inline warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentFunctionsTest.java:240]: Comment "All terms share index 252"

File: warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentFunctionsTest.java [hunk 4]
```diff
@@ -180,6 +180,86 @@ public void testEvaluation1() {
...
```
Comment (apmoriarty) [inline warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentFunctionsTest.java:260]: Comment "All terms share index 252".

Do these last two cases actually exercise the forward/reverse matching logic? The offsets appear to be identical.

**Review by apmoriarty — CHANGES_REQUESTED** [review]

**Review by apmoriarty — APPROVED** [review]

**Review by ivakegg — APPROVED** [review]

**Pull request merged**